### ISSUE TITLE
Wazuh 4.3.0 ignores local_rules.xml

### ISSUE THREAD

**Issue by meganie** [wazuh/wazuh#13407]
|Wazuh version|Component|
|---|---|
| 4.3.0 | Rules | 

After the Update from 4.2.6 to 4.3.0 Wazuh doesn't load my local_rules.xml anymore from /var/ossec/etc/rules/

Here is a rule that used to work before but even if I only have this single rule in the file now it doesn't work.

```
<!-- Full Log for Rule 62123 -->
<group name="windows,windows_defender,">
  <rule id="62123" level="12" overwrite="yes">
    <if_sid>62101</if_sid>
    <field name="win.system.eventID">^1116$</field>
    <description>Windows Defender: Antimalware platform detected $(win.eventdata.severityName) potentially unwanted software ($(win.eventdata.processName))</description>
    <group>pci_dss_5.1,pci_dss_5.2,pci_dss_10.6.1,pci_dss_11.4,gpg13_4.2,gdpr_IV_35.7.d,hipaa_164.312.b,nist_800_53_SI.3,nist_800_53_AU.6,nist_800_53_SI.4,tsc_A1.2,tsc_CC7.2,tsc_CC7.3,tsc_CC6.1,tsc_CC6.8,</group>
  </rule>
</group>
```

**Comment by JcabreraC** [comment]
Hi @meganie,

In previous versions of Wazuh, the bug reported in issue [#8210](https://github.com/wazuh/wazuh/pull/8210), with the overwrite option, was present. To avoid these problems, now, a rule with the overwrite option does not allow the `if_sid`, `if_group` or `if_level` options.

You could see more details in issue [Custom changes to the ruleset not working properly #7948](https://github.com/wazuh/wazuh/issues/7948) and PR [Fix the working of the overwrite option of the rules #8210](https://github.com/wazuh/wazuh/pull/8210).

There are two cases for replacing a rule:
- The rule does not match and should
- The rule matches and should not

The first case is the simplest, since we should create new rules without changing the old ones.
The second case may be more complex. If the rule matches and it should not, you could create a child rule instead of overwriting it to add new conditions, change the level or the description. For example, instead of replacing [rule 60611](https://github.com/wazuh/wazuh/blob/c66924add18180fbf9629ae72e34129b6b7da277/ruleset/rules/0600-win-wdefender_rules.xml#L239-L245), we could create a new child rule of this one:
```xml
<rule id="100011" level="12">
  <if_sid>62123</if_sid>
  <field name="win.system.eventID">^1116$</field>
  <description>Windows Defender: Antimalware platform detected $(win.eventdata.severityName) potentially unwanted software ($(win.eventdata.processName))</description> <group>pci_dss_5.1,pci_dss_5.2,pci_dss_10.6.1,pci_dss_11.4,gpg13_4.2,gdpr_IV_35.7.d,hipaa_164.312.b,nist_800_53_SI.3,nist_800_53_AU.6,nist_800_53_SI.4,tsc_A1.2,tsc_CC7.2,tsc_CC7.3,tsc_CC6.1,tsc_CC6.8,</group>
</rule>
```
Regards !

**Comment by meganie** [comment]
Thank you very much for the explanation!

**Comment by meganie** [comment]
I was able to test it today but if I try to fix [rule 92656](https://github.com/wazuh/wazuh/issues/13393) with my own child rule I still get 92565 events all the time and my own rule doesn't get triggered.

`<group name="win_evt_channel,">
  <rule id="100003" level="15">
    <if_sid>92656</if_sid>
    <field name="win.eventdata.logonType" type="pcre2">10</field>
    <field name="win.eventdata.ipAddress" type="pcre2">::1|127\.0\.0\.1</field>
    <description>User: $(win.eventdata.subjectDomainName)\$(win.eventdata.targetUserName) logged using Remote Desktop Connection (RDP) from loopback address, possible exploit over reverse tunneling using stolen credentials.</description>
    <mitre>
      <id>T1021.001</id>
      <id>T1078.002</id>
    </mitre>
  </rule>
</group>`

**Comment by JcabreraC** [comment]
Hello @meganie,

This problem is being fixed for a minor version to be released soon. You can see the change here: https://github.com/wazuh/wazuh/pull/13409

Your custom rule does not appear because you must set the `logontype` to rule 92656, otherwise, it will generate the bad alert.

I recommend you to edit directly the rule `92656`. To do this, go to the file `/var/ossec/ruleset/rules/0840-win_event_channel.xml` and set the rule like this:

```xml
  <rule id="92656" level="15">
    <if_sid>60106</if_sid>
    <field name="win.eventdata.logonType" type="pcre2">^10$</field>
    <field name="win.eventdata.ipAddress" type="pcre2">::1|127\.0\.0\.1</field>
    <description>User: $(win.eventdata.subjectDomainName)\$(win.eventdata.targetUserName) logged using Remote Desktop Connection (RDP) from loopback address, possible exploit over reverse tunneling using stolen credentials.</description>
    <mitre>
      <id>T1021.001</id>
      <id>T1078.002</id>
    </mitre>
</rule>
```
which is how the rule looks in the following version.

Regards !

**Comment by davidjiglesias** [comment]
Related with #13414

**Comment by JcabreraC** [comment]
Hello @vP3nguin,

The problem with adding fields to a child rule is that evaluating the rule but not matching will generate the alert of the previous rule (the parent rule).

I understand that you only want to receive alerts when `win.eventdata.targetUserName` is `XXXX`. In this case, you should create a child rule of level 0 so that it does not generate an alert:
```xml
<rule id="260122" level="0">
    <if_sid>60122</if_sid>
    <field name="win.system.eventID">^529$|^4625$</field>
    <options>no_full_log</options>
    <description>Logon failure - Unknown user or bad password.</description>
    <mitre>
      <id>T1078</id>
      <id>T1531</id>
    </mitre>
    <group>authentication_failed,gdpr_IV_32.2,gdpr_IV_35.7.d,gpg13_7.1,hipaa_164.312.b,nist_800_53_AC.7,nist_800_53_AU.14,pci_dss_10.2.4,pci_dss_10.2.5,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,</group>
  </rule>
```

And then a child rule of this with that particular field, so that only those alerts are generated:
```xml
<rule id="260123" level="5">
    <if_sid>260122</if_sid>
    <field name="win.system.eventID">^529$|^4625$</field>
    <field name="win.eventdata.targetUserName" type="pcre2">XXXX</field>
    <options>no_full_log</options>
    <description>Logon failure - Unknown user or bad password.</description>
    <mitre>
         <id>T1078</id>
         <id>T1531</id>
    </mitre>
    <group>authentication_failed,gdpr_IV_32.2,gdpr_IV_35.7.d,gpg13_7.1,hipaa_164.312.b,nist_800_53_AC.7,nist_800_53_AU.14,pci_dss_10.2.4,pci_dss_10.2.5,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,</group>
</rule>
```

On the other hand, the `overwrite` problem with `if_sid` has been fixed. You can see the PR here: https://github.com/wazuh/wazuh/pull/13439
This is fixed in version 4.3.1 which will be released these days.

When you have this version, you can reduce the number of rules by overwriting the `60122` rule with `level 0` so that it does not generate alerts and creating a single child rule when `win.eventdata.targetUserName` is `XXXX`.

Please, if you have any questions in the future, please open your own issue instead of using another user's issue. This way the information will be more organized.

Remember that you can contact us in the following channels:
- [Github](https://github.com/wazuh/wazuh/issues/new/choose)
- [Slack](https://join.slack.com/t/wazuh/shared_invite/zt-weqjjmw3-Md50_N3q7PKglVXB3TsgXw)
- [Wazuh mail list](https://groups.google.com/u/0/g/wazuh)

Regards !
